datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -846,6 +861,24 @@ pub async fn from_substrait_rel(
     }
 }
 
+fn requalify_sides_if_needed(
```

Comment:
Perhaps you could add some of the excellent context from this PR's description as docstrings here to help future readers understand what is going on 

This is not strictly necessary as I think the comments in the function explain it, but docstrings would still be nice

Reply:
Added a short version in https://github.com/apache/datafusion/pull/11049/commits/87829ece5f0473c001a56ecd6355b95436751fe1